### PR TITLE
Added support for broken multipart files

### PR DESCRIPTION
**Pull request by moishce**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
https://jira-dc.paloaltonetworks.com/browse/XSUP-30934

## Description
Fixed an issue where EML multipart files were not parsed if they had a broken boundary, which appears only once, and we get an error, so I added a function that deletes the wrong boundary and represse again.

## Screenshots
Paste here any images that will help the reviewer

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: parse_emails/handle_eml.py [hunk 1]
```diff
@@ -79,6 +79,9 @@ def handle_eml(file_path, b64=False, file_name=None, parse_only_headers=False, m
                 headers_map[item[0]] = value
 
         eml = message_from_string(file_data)
+        if eml.defects:
+            eml = handle_multi_part_error(eml)
+
```
Comment (ShahafBenYakir) [inline parse_emails/handle_eml.py:84]: Let's put these two in one function called `create_message_from_string`

File: parse_emails/handle_eml.py [hunk 2]
```diff
@@ -273,6 +276,27 @@ def handle_eml(file_path, b64=False, file_name=None, parse_only_headers=False, m
...
+    Returns:
+        the eml parse obj (Message)
+    """
+    for defect in eml.defects:
+        if isinstance(defect, errors.MultipartInvariantViolationDefect):
+            boundary = eml.get_boundary()
+            file_data = eml.as_string()
+            if file_data.count(boundary) == 1:
+                param = eml.get("Content-Type").replace('\r\n', '\n')
+                file_data = eml.as_string().replace(f'Content-Type: {param}\n', '')
```
Comment (ShahafBenYakir) [inline parse_emails/handle_eml.py:294]:
```suggestion
                file_data = file_data.replace(f'Content-Type: {param}\n', '')
```

File: parse_emails/handle_eml.py [hunk 3]
```diff
@@ -273,6 +276,27 @@ def handle_eml(file_path, b64=False, file_name=None, parse_only_headers=False, m
...
+                eml = message_from_string(file_data)
+            break
```
Comment (ShahafBenYakir) [inline parse_emails/handle_eml.py:296]: why break? it can't be more than once?

**Review by ShahafBenYakir — APPROVED** [review]
looks ok

File: parse_emails/handle_eml.py [hunk 4]
```diff
@@ -273,6 +275,34 @@ def handle_eml(file_path, b64=False, file_name=None, parse_only_headers=False, m
         return email_data, attached_emails
 
 
+def create_message_from_string(file_data):
```
Comment (ShahafBenYakir) [inline parse_emails/handle_eml.py:278]:
```suggestion
def create_message_from_string(file_data: str) -> Message:
```
Let's also add a short docstring

File: parse_emails/handle_eml.py (hunk 3 above)
Comment (ShahafBenYakir) [reply parse_emails/handle_eml.py:302]: @moishce what about this?

**Review by ShahafBenYakir — COMMENTED** [review]
See myt comments

File: parse_emails/handle_eml.py (hunk 3 above)
Comment (moishce) [reply parse_emails/handle_eml.py:302]: It just goes through the list of errors returned, and we check if the error we are looking for is there
if we find it, we can stop

**Review by ShahafBenYakir — APPROVED** [review]

**Pull request merged**